src/js/game/GameController.js

```diff
@@ -1173,6 +1173,7 @@ class GameController {
     let frontBlock = this.levelModel.actionPlane.getBlockAt(frontPosition);
 
     const isFrontBlockDoor = frontBlock === undefined ? false : frontBlock.blockType === "door";
+    const isFrontBlockRails = frontBlock === undefined ? false : frontBlock.blockType.startsWith("rails");
```

Comment:
shouldn't this be `frontBlock.isRail`, particularly since `railsRedstoneTorch` exists as a block type

Reply:
Yeah we can change that. I was merely mimicking the "isFrontBlockDoor" definition there.